### PR TITLE
Bugfix: autocomplete support for common  parameters in path

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:

Autocomplete of 3scale data is failing in some OAS specs because of the following reason:
The `parameters` can be placed in several places:

In the path object, like:
```
"paths": {
  "/": {
    "parameters": [],
    "get": {},
    "post": {}
    ...
  }
}
```
Inside the http verbs, like:
```
"paths": {
  "/": {
    "get": {
      "parameters": []
    },
    "post": {}
    ...
  }
}
```

Or inside both the path and the verb, like:
```
"paths": {
  "parameters": [],
  "/": {
    "get": {
      "parameters": []
    },
    "post": {}
    ...
  }
}
```
This PR adds support for the 3 scenarios

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-6088

**Verification steps** 

Go to API > ActiveDocs
Create a new valid OAS 3 spec (or better use the [default Echo API](https://github.com/3scale/porta/blob/master/public/echo-api-3.0.json), which includes the `x-data-threescale-name` fields)
Test your spec in both Admin and Developer Portal.

oas
____
oas2

**Special notes for your reviewer**:
Related docs:
https://swagger.io/specification/
https://swagger.io/docs/specification/describing-parameters/#path-parameters
Official examples of OAS v3:
https://github.com/OAI/OpenAPI-Specification/tree/master/examples/v3.0

### CODE REVIEW

File: app/javascript/packs/OAS3Autocomplete.js [hunk 1]
```diff
@@ -32,41 +34,40 @@ const addAutocompleteToParam = (param: Param, accountData: AccountData): Param =
...
-        : path[item].parameters
-    }
-  ), {})
-)
-
-const injectParametersField = (key, value, accountData) => {
-  const parametersArray = value.map(param => X_DATA_ATTRIBUTE in param ? addAutocompleteToParam(param, accountData) : param)
-  return (key === 'parameters') ? parametersArray : { ...value, parameters: parametersArray }
+const injectParametersToPathOperation = (pathOperation: PathOperationObject, accountData: AccountData): PathOperationObject => {
+  const operationParameters = pathOperation.parameters || undefined
```
Comment (josemigallas) [inline app/javascript/packs/OAS3Autocomplete.js:38]: I think the second undefined is redundant:
```suggestion
  const operationParameters = pathOperation.parameters
```

File: app/javascript/packs/OAS3Autocomplete.js [hunk 2]
```diff
@@ -32,41 +34,40 @@ const addAutocompleteToParam = (param: Param, accountData: AccountData): Param =
...
+  if (!operationParameters) {
+    return pathOperation
+  }
+  const parametersWithAutocompleteData = operationParameters.map(param => X_DATA_ATTRIBUTE in param ? addAutocompleteToParam(param, accountData) : param)
+  return {
+    ...pathOperation,
+    'parameters': parametersWithAutocompleteData
```
Comment (josemigallas) [inline app/javascript/packs/OAS3Autocomplete.js:45]: No need for quotes:
```suggestion
    parameters: parametersWithAutocompleteData
```

File: app/javascript/packs/OAS3Autocomplete.js [hunk 3]
```diff
@@ -32,41 +34,40 @@ const addAutocompleteToParam = (param: Param, accountData: AccountData): Param =
...
+  param => X_DATA_ATTRIBUTE in param ? addAutocompleteToParam(param, accountData) : param
+)
+
+const injectParametersToPath = (path: PathItemObject, commonParameters: Array<Param>, accountData: AccountData): PathItemObject => (
+  Object.keys(path).reduce((updatedPath, item) => {
+    if (item === 'parameters' && commonParameters) {
+      updatedPath[item] = injectAutocompleteToCommonParameters(commonParameters, accountData)
+    } else {
+      updatedPath[item] = injectParametersToPathOperation(path[item], accountData)
+    }
```
Comment (josemigallas) [inline app/javascript/packs/OAS3Autocomplete.js:59]:
```suggestion
    updatedPath[item] = (item === 'parameters' && commonParameters)
      ? injectAutocompleteToCommonParameters(commonParameters, accountData)
      : injectParametersToPathOperation(path[item], accountData)
    return updatedPath
```

File: app/javascript/packs/OAS3Autocomplete.js (hunk 1 above)
Comment (josemigallas) [inline app/javascript/packs/OAS3Autocomplete.js:38]:
```suggestion
  const operationParameters = pathOperation.parameters
```

File: app/javascript/packs/OAS3Autocomplete.js (hunk 2 above)
Comment (josemigallas) [inline app/javascript/packs/OAS3Autocomplete.js:45]:
```suggestion
    parameters: parametersWithAutocompleteData
```

File: app/javascript/packs/OAS3Autocomplete.js [hunk 4]
```diff
@@ -32,41 +34,40 @@ const addAutocompleteToParam = (param: Param, accountData: AccountData): Param =
...
+    return updatedPath
```
Comment (josemigallas) [inline app/javascript/packs/OAS3Autocomplete.js:60]:
```suggestion
    updatedPath[item] = (item === 'parameters' && commonParameters)
      ? injectAutocompleteToCommonParameters(commonParameters, accountData)
      : injectParametersToPathOperation(path[item], accountData)
    return updatedPath
```

File: app/javascript/packs/OAS3Autocomplete.js [hunk 5]
```diff
@@ -32,41 +34,40 @@ const addAutocompleteToParam = (param: Param, accountData: AccountData): Param =
...
-)
-
-const injectParametersField = (key, value, accountData) => {
-  const parametersArray = value.map(param => X_DATA_ATTRIBUTE in param ? addAutocompleteToParam(param, accountData) : param)
-  return (key === 'parameters') ? parametersArray : { ...value, parameters: parametersArray }
+const injectParametersToPathOperation = (pathOperation: PathOperationObject, accountData: AccountData): PathOperationObject => {
+  const operationParameters = pathOperation.parameters || undefined
+  if (!operationParameters) {
+    return pathOperation
+  }
```
Comment (didierofrivia) [inline app/javascript/packs/OAS3Autocomplete.js:41]:
```suggestion
  if (!operationParameters) return pathOperation
```

**Review by didierofrivia — APPROVED** [review]

**Review by josemigallas — APPROVED** [review]

**Pull request merged**